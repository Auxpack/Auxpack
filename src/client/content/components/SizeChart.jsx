import React, { Component } from 'react';
import * as d3 from 'd3';
// import * as d3 from "d3"; << NOT IMPORTING, AS WE'VE BUILT ON d3.v4, as imported in index.html

class SizeChart extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        //RENDERING CHART WHEN COMPONENT MOUNTS
        this.drawChart();
    }

    drawChart() {
        //CHART MARGINS AND DIMENSIONS
        var margin = {top: 10, right: 30, bottom: 50, left: 60},
            width = 350 - margin.left - margin.right,
            height = 300 - margin.top - margin.bottom;

        //APPENDING THE OVERALL SVG CHART
        var svg = d3.select("#size-chart")
        .append("svg")
            .attr("width", width + margin.left + margin.right)
            .attr("height", height + margin.top + margin.bottom)
            .attr("class", "build-chart-svg")
        .append("g")
            .attr("transform",
                "translate(" + margin.left + "," + margin.top + ")");

        //TAKING IN THE PASSED DOWN DATA
        let chartData = this.props.chartData;

        //IFFY FUNCTION TO IMMEDIATELY RENDER OUT THE CHART
        (function() {
            //ADDING X-AXIS - BUILD
            var x = d3.scaleTime()
            .domain(d3.extent(chartData, function(d) { return d.build; }))
            .range([ 0, width ]);
            svg.append("g")
            .attr("transform", "translate(0," + height + ")")
            .call(d3.axisBottom(x).tickFormat(d3.format("d")));

            //ADDING Y-AXIS - SIZE
            var y = d3.scaleLinear()
            .domain([0, d3.max(chartData, function(d) { return +d.size; })])
            .range([ height, 0 ]);
            svg.append("g")
            .call(d3.axisLeft(y));

            //ADDING THE LINE
            svg.append("path")
            .datum(chartData)
            .attr("fill", "none")
            .attr("stroke", "rgb(63,81,181)")
            .attr("stroke-width", 1.8)
            .attr("d", d3.line()
                .x(function(d) { return x(d.build) })
                .y(function(d) { return y(d.size) })
                )

            //ADDING X-AXIS LABEL
            svg.append("text")
            .attr("class", "x label")
            .attr("text-anchor", "end")
            .attr("x", width)
            .attr("y", height + 35)
            .text("build, by size")
            .attr("fill", "rgba(63,81,181,1)");

            //ADDING Y-AXIS LABEL
            svg.append("text")
            .attr("class", "y label")
            .attr("text-anchor", "end")
            .attr("y", -50)
            .attr("dy", ".75em")
            .attr("transform", "rotate(-90)")
            .text("size\xa0\xa0\xa0\xa0(in kB)")
            .attr("fill", "rgba(63,81,181,1)");

        })()
    }
    
    render() {
        return (
            <div className="single-chart">
                <div id="size-chart"></div>
            </div>
        )
    }
}

export default SizeChart;