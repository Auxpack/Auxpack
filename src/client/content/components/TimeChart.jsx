import React, { Component } from 'react';
// import * as d3 from "d3";

class TimeChart extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        this.drawChart();
    }

    drawChart() {
        // set the dimensions and margins of the graph
        var margin = {top: 10, right: 30, bottom: 30, left: 60},
            width = 300 - margin.left - margin.right,
            height = 250 - margin.top - margin.bottom;

        // append the svg object to the body of the page
        var svg = d3.select("#time-chart")
        .append("svg")
            .attr("width", width + margin.left + margin.right)
            .attr("height", height + margin.top + margin.bottom)
        .append("g")
            .attr("transform",
                "translate(" + margin.left + "," + margin.top + ")");

        let chartData = this.props.chartData;

        //Read the data
        d3.csv("https://raw.githubusercontent.com/holtzy/data_to_viz/master/Example_dataset/3_TwoNumOrdered_comma.csv",

        // Now I can use this dataset:
        function(data) {
            // Add X axis --> it is a date format
            var x = d3.scaleTime()
            .domain(d3.extent(chartData, function(d) { return d.build; }))
            .range([ 0, width ]);
            svg.append("g")
            .attr("transform", "translate(0," + height + ")")
            .call(d3.axisBottom(x));

            // Add Y axis
            var y = d3.scaleLinear()
            .domain([0, d3.max(chartData, function(d) { return +d.time; })])
            .range([ height, 0 ]);
            svg.append("g")
            .call(d3.axisLeft(y));

            // Add the line
            svg.append("path")
            .datum(chartData)
            .attr("fill", "none")
            .attr("stroke", "rgb(63,81,181)")
            .attr("stroke-width", 1.5)
            .attr("d", d3.line()
                .x(function(d) { return x(d.build) })
                .y(function(d) { return y(d.time) })
                )

        })
    }

    render() {
        return (
            <div className="single-chart">
                <div id="time-chart"></div>
                <p className="chart-label">TIME CHART</p>
            </div>
        )
    }
}

export default TimeChart;