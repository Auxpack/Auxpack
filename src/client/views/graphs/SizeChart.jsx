// import React, { Component } from 'react';
// import * as d3 from "d3";

// const SizeGraph = ({ chartData }) => {

//     var margin = { top: 10, right: 30, bottom: 30, left: 60 },
//         width = 460 - margin.left - margin.right,
//         height = 400 - margin.top - margin.bottom;

//     // append the svg object to the body of the page
//     var svg = d3.select("#sizeGraph")
//         .append("svg")
//         .attr("width", width + margin.left + margin.right)
//         .attr("height", height + margin.top + margin.bottom)
//         .append("g")
//         .attr("transform",
//             "translate(" + margin.left + "," + margin.top + ")");

//     //Read the data
//     d3.csv("https://raw.githubusercontent.com/holtzy/data_to_viz/master/Example_dataset/3_TwoNumOrdered_comma.csv",
//         // When reading the csv, I must format variables:
//         function (d) {
//             return { date: d3.timeParse("%Y-%m-%d")(d.date), value: d.value }
//         },
//         // Now I can use this dataset:
//         function (data) {
//             // Add X axis --> it is a date format
//             var x = d3.scaleTime()
//                 .domain(d3.extent(data, function (d) { return d.date; }))
//                 .range([0, width]);
//             svg.append("g")
//                 .attr("transform", "translate(0," + height + ")")
//                 .call(d3.axisBottom(x));

//             // Add Y axis
//             var y = d3.scaleLinear()
//                 .domain([0, d3.max(data, function (d) { return +d.value; })])
//                 .range([height, 0]);
//             svg.append("g")
//                 .call(d3.axisLeft(y));

//             // Add the line
//             svg.append("path")
//                 .datum(data)
//                 .attr("fill", "none")
//                 .attr("stroke", "steelblue")
//                 .attr("stroke-width", 1.5)
//                 .attr("d", d3.line()
//                     .x(function (d) { return x(d.date) })
//                     .y(function (d) { return y(d.value) })
//                 )

//         })
//     return (
//         <div id="sizeChart" className="chart">
//             <svg width={630} height={500} className="#chart" ref={(elem) => { this.svg = elem; }} />
//         </div>
//     )
// }

// export default SizeGraph;