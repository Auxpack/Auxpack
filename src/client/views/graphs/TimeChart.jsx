import React, { Component } from 'react';
import * as d3 from "d3";

const TimeGraph = ({ chartData }) => {

    return <React.Fragment>
        {chartData[0].build}
    </React.Fragment>
}

export default TimeGraph;