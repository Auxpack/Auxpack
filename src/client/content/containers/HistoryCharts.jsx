import React, { useState, useEffect } from 'react';
import SizeChart from '../components/SizeChart.jsx';
import TimeChart from '../components/TimeChart.jsx';

const HistoryCharts = ({ build }) => {

    //PARSING PASSED DOWN BUILD DATA INTO SIMPLE OBJECTS FOR CHARTS

    let chartData = build.map((b, i) => {
        return b = {
            'build': i + 1,
            'size': b.size / 1000,
            'time': b.time / 1000,
        }
    })

    return <React.Fragment>
            <div id="chart-container">
                <SizeChart chartData={chartData} />
                <TimeChart chartData={chartData} />
            </div>
        </React.Fragment>
}

export default HistoryCharts;