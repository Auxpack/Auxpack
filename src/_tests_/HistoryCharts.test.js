import HistoryCharts from '../client/content/containers/HistoryCharts.jsx';
import SizeChart from '../client/content/components/SizeChart.jsx';
import TimeChart from '../client/content/components/TimeChart.jsx';

describe('HistoryCharts Unit Testing', () => {
  describe('HistoryCharts Container', () => {
    let wrapper;

    const props = {
      build: [{
        timeStamp: 1575426090404,
        time: 1439,
        hash: '546142ce1b49a6ba7d6f',
        errors: [],
        size: 1172113,
        assets: [{ name: 'bundle.js', chunks: ['main'], size: 1172113 }],
        chunks: [{ size: 1118609, files: ['bundle.js'], modules: [{ name: './client/App.jsx', size: 6375, id: './client/App.jsx' }] }],
        treeStats: {
          cjs: [{ name: './test2.js', size: 49, reasonTypes: [{ name: './test.js', type: 'cjs require' }] }, { name: './test2.js', size: 49, reasonTypes: [{ name: './test.js', type: 'cjs require' }] }],
          esm: [{ name: './client/App.jsx', size: 6375, reasonTypes: [{ name: './client/index.js', type: 'harmony side effect evaluation' }] }],
          both: [{ name: './node_modules/react/index.js', size: 190, reasonTypes: [{ name: './client/App.jsx', type: 'harmony side effect evaluation' }] }],
        },
      }],
    };

    beforeAll(() => {
      wrapper = shallow(<HistoryCharts {...props} />);
    });

    it('HistoryCharts snapshot testing', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('HistoryCharts should render SizeChart and TimeChart', () => {
      expect(wrapper.find(SizeChart).length).toBe(1);
      expect(wrapper.find(TimeChart).length).toBe(1);
    });
  });

  describe('SizeChart', () => {
    let wrapper;

    const props = {
      chartData: [{ build: 1, size: 100, time: 3 }, { build: 2, size: 100, time: 3 }, { build: 3, size: 100, time: 3 }],
    };

    beforeAll(() => {
      wrapper = mount(<SizeChart {...props} />);
    });

    it('SizeChart snapshot testing', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('SizeChart should render divs with class "single-chart" and id "size-chart"', () => {
      expect(wrapper.find('div').at(0).hasClass('single-chart')).toBe(true);
      expect(wrapper.find('#size-chart').length).toBe(1);
    });
  })

  describe('TimeChart', () => {
    let wrapper;

    const props = {
      chartData: [{build: 1, size: 100, time: 3}, {build: 2, size: 100, time: 3}, {build: 3, size: 100, time: 3}]
    }

    beforeAll(() => {
      wrapper = mount(<TimeChart {...props} />)
    });

    it('TimeChart snapshot testing', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('TimeChart should render divs with class "single-chart" and id "time-chart"', () => {
      expect(wrapper.find('div').at(0).hasClass("single-chart")).toBe(true);
      expect(wrapper.find('#time-chart').length).toBe(1);
    });
  })
})
