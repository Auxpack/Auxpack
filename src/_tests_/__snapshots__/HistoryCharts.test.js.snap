// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryCharts Unit Testing HistoryCharts Container HistoryCharts snapshot testing 1`] = `
<Fragment>
  <div
    className="content-card"
  >
    <div
      id="chart-container"
    >
      <SizeChart
        chartData={
          Array [
            Object {
              "build": 1,
              "size": 1172.113,
              "time": 1.439,
            },
          ]
        }
      />
      <TimeChart
        chartData={
          Array [
            Object {
              "build": 1,
              "size": 1172.113,
              "time": 1.439,
            },
          ]
        }
      />
    </div>
  </div>
</Fragment>
`;

exports[`HistoryCharts Unit Testing SizeChart SizeChart snapshot testing 1`] = `
<SizeChart
  chartData={
    Array [
      Object {
        "build": 1,
        "size": 100,
        "time": 3,
      },
      Object {
        "build": 2,
        "size": 100,
        "time": 3,
      },
      Object {
        "build": 3,
        "size": 100,
        "time": 3,
      },
    ]
  }
>
  <div
    className="single-chart"
  >
    <div
      id="size-chart"
    />
  </div>
</SizeChart>
`;

exports[`HistoryCharts Unit Testing TimeChart TimeChart snapshot testing 1`] = `
<TimeChart
  chartData={
    Array [
      Object {
        "build": 1,
        "size": 100,
        "time": 3,
      },
      Object {
        "build": 2,
        "size": 100,
        "time": 3,
      },
      Object {
        "build": 3,
        "size": 100,
        "time": 3,
      },
    ]
  }
>
  <div
    className="single-chart"
  >
    <div
      id="time-chart"
    />
  </div>
</TimeChart>
`;
